src/api/general-services/pipeline-manage/index.js

```diff
@@ -249,7 +249,11 @@ const createGem2SPipeline = async (experimentId) => {
 
   if (metadataKeys.length) {
     taskParams.metadata = metadataKeys.reduce((acc, key) => {
-      acc[key] = samples.ids.map(
+      // Make sure the key does not contain '-' as it will cause failure in GEM2S
+      const sanitizedKey = key.replace(/-+/g, '_');
+
+      acc[sanitizedKey] = samples.ids.map(
+        // Fetch using unsanitized key as it is the key used to store metadata in sample
         (sampleUuid) => samples[sampleUuid].metadata[key] || defaultMetadataValue,
```

Comment:
Shouldn't we be returning `acc`? (I hate reduce, BTW :-D )

Reply:
It's returned on line 259, you have to scroll down a bit to see it.

Reply:
It takes sometime to get to know `reduce`, but once you get the hang of it, it's pretty powerful (albeit difficult to read)

Reply:
Oh, sorry! Thanks!